yappi/_yappi.c

```diff
@@ -1147,6 +1207,47 @@ _yapp_callback(PyObject *self, PyFrameObject *frame, int what,
     return 0;
 }
 
+static void
+_pause_greenlet_ctx(_ctx *ctx)
+{
+    ydprintf("pausing context: %ld %s\n", ctx->id, ctx->name);
+    ctx->paused = 1;

```

Comment:
Just an cosmetic idea: how about putting greenlet specific variables into something like ctx->greenlet_data or `ctx->greenlet` something. Since we might have different ctx backends share this same structure if would be nice which variables belong to which backend conceptually.